### DISCUSSION TITLE
How to cusomize RUNNER_WORKDIR ?

### DISCUSSION THREAD

**Discussion by vietwow** [actions/actions-runner-controller#2127] [Questions]
Hi all,

I need to customize the working dir of action runner, I tried to set env var "RUNNER_WORKDIR" in controller :

> apiVersion: actions.summerwind.dev/v1alpha1
kind: RunnerDeployment
metadata:
  name: golang-runner
  namespace: platform
spec:
  replicas: 3
  template:
    spec:
      env:
      - name: RUNNER_WORKDIR
        value: /home/runner
      repository: dtduc-git/golang
      labels:
        - golang-runner
      serviceAccountName: gha

but it doesn't work, when I describe the runner's pod, I see 2 duplicated env var "RUNNER_WORKDIR", and runner is still using the value of old "RUNNER_WORKDIR"

so how could I customize working dir of action runner?

Appricated for any help. Thank you.

Best Regards,

**Comment by jbkc85** [comment]
@vietwow I am not sure you should be setting the ENV variable.  It is actually written in there with the RunnerDeployment spec:

https://github.com/actions/actions-runner-controller/blob/044c8ad4d5d48e50e76f88340dd45511167ee69d/controllers/actions.summerwind.net/runner_controller.go#L575

So, when you are setting your RunnerDeployment spec, just set up the `workDir` to be your desired home:

```
  workDir: /home/to/somewhere/else
```
